### ISSUE TITLE
Issue in running G2 Classes from Tomcat

### ISSUE THREAD

**Issue by samuelebe** [Senzing/senzing-api-server#123]
**Describe the bug**
I have been trying to set this project in eclipse and got stuck in executing in server. I thought I understand senzing setup process based on the knowledge docs and followed all the steps as it mentioned. Not sure what and where I am doing wrong. Can you guys please help me in fixing this and I am going to document step by step procedure so that I wont need to do it all over again every time 
1)	Installed Senzing under /opt/senzing as “root” user
2)	Created senzing project under /home/sila_ey_user/senzing under “sila_ey_user” account
3)	Installed Apache tomcat 9 as “root” user under /opt/tomcat/
4)	Deployed SenzingApp.war in Tomcat webapps as “tomcat” user
a.	SenzingApp.war – This application basically connects to Senzing and search for given identity based on his/her attributes.
5)	Set senzing parameters, tomcat and java library path in .bashrc and .bash_profile for all “root, tomcat and sila_ey_user” accounts
6) Make sure that g2.jar and libG2.so is available in java library path
7)  Start tomcat
8) Run the url in browser which calls the Java file in SenzingApp application

1. Go to Tomcat localhost and hit the application url

**Expected behavior**
1) Loads G2.so and all Jars properly
2) Render the service required from Java class file in Tomcat Application

![screenshot1](https://user-images.githubusercontent.com/28157977/72375496-34831e00-36da-11ea-8cc0-cdd76fdda4f0.jpg)
![screenshot2](https://user-images.githubusercontent.com/28157977/72375498-351bb480-36da-11ea-8ea5-fc28070dc1d2.jpg)
![screenshot3](https://user-images.githubusercontent.com/28157977/72375499-351bb480-36da-11ea-81f7-a8f404c4b421.jpg)
![screenshot4](https://user-images.githubusercontent.com/28157977/72375500-351bb480-36da-11ea-96b8-ac85ad84cb58.jpg)
![screenshot5](https://user-images.githubusercontent.com/28157977/72375501-35b44b00-36da-11ea-8fcf-a26d300ccb63.jpg)

**Additional context**

**Comment by barrycaceres** [comment]
@samuelebe Can we do a screen share with Zoom and I can maybe walk you through fixing the issue?

When are you available?

**Comment by barrycaceres** [comment]
I am pretty sure it is an environment variable.  The following must be set in the shell where the API server code is run:
- "LD_LIBRARY_PATH" (Linux)"DYLD_LIBRARY_PATH" (macOS) or "Path" (Windows)

**Comment by samuelebe** [comment]
@barrycaceres 
I am able to load APIs now, but as soon as I hit the URL, it gives below exception and stops tomcat server. Error says it cannot load database type sqlite3. Is this something to do with V1 APIs?

```
2020-01-15 16:25:29.392 [G2JNI:140079087363840] DBUG: Shards are not used
SQLITE3 ERROR: ((14:unable to open database file  unable to open database file ))
2020-01-15 16:25:29.392 [sql:140079087363840:[1::MASTER::]] CRIT: Exception: unable to open database file  unable to open database file
2020-01-15 16:25:29.392 [sql:140079087363840:[1::MASTER::]] CRIT: Statement: Statement: query = SELECT UTYPE_CODE FROM LIB_UTYPE
2020-01-15 16:25:29.392 [sql:140079087363840:[1::MASTER::]] CRIT: No Bind Values
g2ModuleInit:-2
ERROR: Couldn't init engine, non-zero return code: -2

2020-01-15 16:25:29.393 [G2JNI:140079087363840] NOTE: TRACE: G2_getLastException([0x7f66b43cf400],[65536])
ERROR: Module Init Exception: 1001E|Critical Database Error '(14:unable to open database file  unable to open database file )'


Initilizing Senzing Config with: /home/sila_ey_user/senzing/etc/G2Module.ini

Initilizing Senzing Config with: /home/sila_ey_user/senzing/etc/G2Module.ini
2020-01-15 16:25:29.394 [G2JNI:140079087363840] NOTE: TRACE: G2Audit_init([G2JNI],[/home/sila_ey_user/senzing/etc/G2Module.ini],[1])
2020-01-15 16:25:29.394 [G2JNI:140079087363840] DBUG: Connecting to database type: sqlite3
2020-01-15 16:25:29.394 [G2JNI:140079087363840] DBUG: Connected to database
2020-01-15 16:25:29.394 [G2JNI:140079087363840] DBUG: Shards are not used
SQLITE3 ERROR: ((14:unable to open database file  unable to open database file ))
2020-01-15 16:25:29.394 [sql:140079087363840:[3::MASTER::]] CRIT: Exception: unable to open database file  unable to open database file
2020-01-15 16:25:29.394 [sql:140079087363840:[3::MASTER::]] CRIT: Statement: Statement: query = SELECT UTYPE_CODE FROM LIB_UTYPE
2020-01-15 16:25:29.394 [sql:140079087363840:[3::MASTER::]] CRIT: No Bind Values
terminate called after throwing an instance of 'SRDSQL_Exception'
  what():  Critical Database Error '(14:unable to open database file  unable to open database file )'

```

**Comment by docktermj** [comment]
@samuelebe Can you post the contents of your `/home/sila_ey_user/senzing/etc/G2Module.ini` file?

**Comment by samuelebe** [comment]
```
[PIPELINE]
 SUPPORTPATH=/home/sila_ey_user/senzing/data
 CONFIGPATH=/home/sila_ey_user/senzing/etc
 RESOURCEPATH=/home/sila_ey_user/senzing/resources

[SQL]
 G2CONFIGFILE=/home/sila_ey_user/senzing/etc/g2config.json
 CONNECTION=sqlite3://na:na@/home/sila_ey_user/senzing/var/sqlite/G2C.db
# CONNECTION=postgresql://username:password@hostname:5432:database/?=schema=schemaname
# CONNECTION=mysql://username:password@hostname:3306/?schema=schemaname
# CONNECTION=db2://username:password@database/?schema=schemaname
# CONNECTION=mssql://username:password@database
```

**Comment by docktermj** [comment]
Just to verify:  The file `/home/sila_ey_user/senzing/var/sqlite/G2C.db` does exist and has ownership and permissions that can be used by your program?

**Comment by samuelebe** [comment]
yes, it has full permissions read and write..

```
/home/sila_ey_user/senzing/var/sqlite
-rwxrwxrwx. 1 sila_ey_user sila_ey_user 12009472 Jan 15 16:32 G2C.db
```

**Comment by docktermj** [comment]
Ok at this point,  I need @barrycaceres' help.

**Comment by brianmacy** [comment]
Also, note that the directory the file is in needs to have read/write permissions.

**Comment by samuelebe** [comment]
yes, the directory has full read and write permissions

**Comment by barrycaceres** [comment]
I'm pretty sure we hit this before and it had to do with the Sqlite native libraries installed on Linux I remember somebody hit it and then they told me they fixed it by updating SQLite -- I am going to ask around and see if that rings any bells so the person who did it can tell me exactly what they fixed.  But to be sure....

**Comment by samuelebe** [comment]
3.7.17 2013-05-20 00:56:22 118a3b35693b134d56ebd780123b7fd6f1497668

**Comment by barrycaceres** [comment]
Let's coordinate another screen share in zendesk and get to the bottom of this.

**Comment by samuelebe** [comment]
sure, can we do it on Monday your preferred time? I am available on coming Monday

**Comment by barrycaceres** [comment]
I will send you an invite and invite a couple of teammates to help figure it out.

**Comment by barrycaceres** [comment]
Let me know in Zendesk what times you are available and I will post a zoom meeting URL in there once we coordinate a time.

**Comment by samuelebe** [comment]
I am okay to join any time between 10AM EST - 4PM EST

**Comment by barrycaceres** [comment]
I set us up for 10AM PST / 1PM PST on Zoom.

The info is in the Zendesk ticket.   Please confirm you have received it and will be able to join.

**Comment by docktermj** [comment]
Just curious, any progress on this one?

**Comment by barrycaceres** [comment]
@samuelebe if you are done with this issue, can you close it out?  If there is something pending on this issue we would love to help you address it.  If a new issue arises we can open a new one.

If I don't hear back from you by the end of the week then I can close it out for you -- but I would not want to do that prematurely.  Let us know -- thanks.

**Comment by samuelebe** [comment]
I was able to get this thing working and we were able to give a demo successfully to client. Thanks for all your help on this project.

**Comment by docktermj** [comment]
Delighted to hear this.  Thank you for helping make the project better!
